tests/frame_test.go

```diff
@@ -72,3 +74,46 @@ func TestFrameDismissDialogBox(t *testing.T) {
 		})
 	}
 }
+
+func TestFrameNoPanicWithEmbeddedIFrame(t *testing.T) {
+	if strValue, ok := os.LookupEnv("XK6_HEADLESS"); ok {
```

Comment:
Just FYI that there's also a test-specific env var for this: [`XK6_BROWSER_TEST_HEADLESS`](https://github.com/grafana/xk6-browser/blob/3cee653dff64cbd71c86c8fe16ebd9828fe3b58b/tests/test_browser.go#L359).

I forget why @inancgumus wanted to use it instead of `XK6_HEADLESS`, but you can discuss it with him. I think this is fine as well.

Reply:
I saw that too, i'll stick with `XK6_HEADLESS` for now 👍

Reply:
IIRC, we decided to prefix environment variables with `XK6_BROWSER` to be explicit and prevent clashes with other extensions.